app/scripts/lib/rpc-method-middleware/createMethodMiddleware.js

```diff
@@ -35,7 +32,18 @@ export default function createMethodMiddleware(opts) {
     const handler = handlerMap.get(req.method);
     if (handler) {
       const { implementation, hookNames } = handler;
-      return implementation(req, res, next, end, selectHooks(opts, hookNames));
+      try {
+        // Implementations may or may not be async, so we must await them.
+        return await implementation(
+          req,
+          res,
+          next,
+          end,
+          selectHooks(hooks, hookNames),
+        );
+      } catch (error) {
+        return end(error);
+      }
```

Comment:
The addition of the `try`/`catch` block fixes a pre-existing issue where errors thrown by async middleware functions would be completely uncaught.

Reply:
Yikes! I wonder how many Sentry errors were caused by this.